File: getting-started-mocks/node/notifier.test.js
import assert from 'node:assert/strict'
import test from 'node:test'

import { notificationFor } from './notifier.js'

test('creates a notification when the package is delayed', () => {
  const getShipmentStatus = () => 'delayed'

  const message = notificationFor(getShipmentStatus, 'TRACK-123')

  assert.equal(message, 'Package TRACK-123 is delayed')
})

test('creates no notification when the package is delivered', () => {
  const getShipmentStatus = () => 'delivered'

  const message = notificationFor(getShipmentStatus, 'TRACK-123')

  assert.equal(message, null)
})

File: getting-started-mocks/go/notifier_test.go
package notifier

import "testing"

func TestNotificationForDelayedPackage(t *testing.T) {
	getShipmentStatus := func(string) string { return "delayed" }

	message, notify := NotificationFor(getShipmentStatus, "TRACK-123")

	if !notify {
		t.Fatal("expected a notification")
	}
	if message != "Package TRACK-123 is delayed" {
		t.Fatalf("unexpected message: %q", message)
	}
}

func TestNotificationForDeliveredPackage(t *testing.T) {
	getShipmentStatus := func(string) string { return "delivered" }

	message, notify := NotificationFor(getShipmentStatus, "TRACK-123")

	if notify {
		t.Fatalf("expected no notification, got %q", message)
	}
}
